### PR TITLE
feat: add fallback mechanism for konnect cloud envs

### PR DESCRIPTION
**Pull request by GGabriele**

### CODE REVIEW

File: cmd/common_konnect.go [hunk 1]
```diff
@@ -33,19 +40,34 @@ func authenticate(ctx context.Context, client *konnect.Client, host string) (kon
...
+	var parsedAddress *url.URL
+	var konnectAddress string
+	for _, address := range addresses {
+		// get Konnect client
+		konnectConfig.Address = address
+		konnectClient, err = utils.GetKonnectClient(httpClient, konnectConfig)
+		if err != nil {
+			return nil, err
+		}
+		parsedAddress, _ = url.Parse(address)
```
Comment (hbagdi) [inline cmd/common_konnect.go:58]: Please handle error.

File: cmd/common_konnect.go [hunk 2]
```diff
@@ -33,19 +40,34 @@ func authenticate(ctx context.Context, client *konnect.Client, host string) (kon
 
 func getKonnectClient(ctx context.Context) (*kong.Client, error) {
```
Comment (hbagdi) [inline cmd/common_konnect.go:41]: getKongClientForKonnectMode, please rename

Comment (hbagdi) [reply cmd/common_konnect.go:41]: Please add a docstring explaining the 'why' behind the code in this function.

**Review by hbagdi — COMMENTED** [review]
Looks good after changes.
Please merge only once #645 is merged in (change base of the PR to main).

File: cmd/common_konnect.go (hunk 2 above)
Comment (GGabriele) [reply cmd/common_konnect.go]: done

**Pull request merged**